addons/website/models/website.py

```diff
@@ -484,12 +484,128 @@ def _OLG_api_rpc(self, route, params):
     def get_cta_data(self, website_purpose, website_type):
         return {'cta_btn_text': False, 'cta_btn_href': '/contactus'}
 
+    def _get_configurator_snippets_addons(self):
+        """Retrieve module specific default configurator snippets addons for theme_default"""
+        return get_manifest('website').get('configurator_snippets_addons', {})
+
+    def _get_snippet_defaults(self, snippet):
+        """Retrieve the default configuration for a given dynamic snippet."""
+        return {}
+
+    def _get_snippet_view_key(self, snippet, page_code):
+        if '.' not in snippet:
+            snippet = 'website.' + snippet
+        module, snippet = snippet.split('.')
+        return f'{module}.configurator_{page_code}_{snippet}'
+
+    def _preconfigure_snippet(self, snippet, el, customizations):
+        """Apply default configuration values to a snippet element.
+
+        This ensures that when a dynamic snippet is appended via the
+        configurator, all of its required default classes/attributes
+        are added to the DOM element before it is rendered.
+        """
+        default_settings = self._get_snippet_defaults(snippet)
+
+        snippet_classes = el.get('class', '').split()
+
+        filter_name = customizations.get('filter_xmlid') or default_settings.get('filter_xmlid')
+        if filter_name:
+            selected_filter = self.env.ref(filter_name)
+            el.set('data-filter-id', str(selected_filter.id))
+            el.set('data-number-of-records', str(selected_filter.limit))
+
+        selected_template_key = customizations.get('template_key') or default_settings.get('template_key')
+        if selected_template_key:
+            el.set('data-template-key', selected_template_key)
+            template_class = re.sub(r'.*\.dynamic_filter_template_', 's_', selected_template_key)
+            if template_class not in snippet_classes:
+                snippet_classes.append(template_class)
+
+        add_classes = customizations.get('add_classes', []) or default_settings.get('add_classes', [])
+        remove_classes = customizations.get('remove_classes', []) or default_settings.get('remove_classes', [])
+
+        if add_classes or remove_classes:
+            for snippet_class in remove_classes:
+                snippet_classes.remove(snippet_class)
+            snippet_classes.extend(add_classes)
+
+        data_attributes = {
+            **default_settings.get('data_attributes', {}),
+            **customizations.get('data_attributes', {}),
+        }
+        for key, value in data_attributes.items():
+            el.set(f'data-{key}', value)
+
+        el.set('class', ' '.join(snippet_classes))
+
+        style_attrs = customizations.get('style_attributes', {}) or default_settings.get('style_attributes', {})
```

Comment:
Nitpick but semantically this is not correct. We are actually setting properties to the `style=""` attribute. I think that `style` instead of `style_attributes` could work fine